fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java

```diff
@@ -842,28 +844,35 @@ public Status resetIdsForRestore(GlobalStateMgr globalStateMgr, Database db, int
         // reset partition info and idToPartition map
         if (partitionInfo.isRangePartition()) {
             RangePartitionInfo rangePartitionInfo = (RangePartitionInfo) partitionInfo;
+            RangePartitionInfo origRangePartitionInfo = (RangePartitionInfo) rangePartitionInfo.clone();
```

Comment:
since all the info need to be reset, is it possible to create a new empty rangePartitionInfo, and put the item into it one by one from the old one with new generated id?

Reply:
or shall we add an interface in partitionInfo, let partitionInfo itself to take care of this swap/replacement meta change, instead of manipulating the meta outside.

Reply:
good suggestion, I will refactor the code in next pr, this is only a bugfix pr